### PR TITLE
feat(uirefresh): component refresh - positions table

### PR DESCRIPTION
**Pull request by moo-onthelawn**
implement positions table changes from figma [here](https://www.figma.com/design/fMQodZKzn9B5aZTN5G0fLJ/dYdX-%E2%80%BA-Web?node-id=41600-21654) 
- did not update filter ui on horizontalPanel, will follow up with that
- did not update fills / order table (plan is to seperate stacked columns into individual ones like positions for visual cohesiveness)
- did no update positions table in mobile tablet view yet - will update in follow up

nit other things:
- replace empty dash with [em dash](https://www.merriam-webster.com/grammar/em-dash-en-dash-how-to-use)
- verified that in testnet, positions table ui hasn't changed
- added truncation to SL/TP columns. It doesn't look great for v small prices but I think it's a reasonable state given the other prices?? (using fixed width 70px on SL/TP columns - I was trying to use flex to make it show the whole number all the time but it was hard to get it to resize properly and stay in sync with the header 🤔 I can follow up to dig in more into this if people feel strongly tho)
- Verified SL/TP box pretty extensively (with liquidation warning, partial order icon, no orders, and multiple orders)
Screenshot 2024-10-16 at 12 19 39 PM
Screenshot 2024-10-16 at 12 19 14 PM

---

## Views

- `pages/trade/HorizontalPanel`
  - Update columns to be shown for `PositionsTable` when `uiRefresh` is on

- `views/tables/PositionsTable/CloseAllPositionsButton`
  - Remove padding for better alignment; checked in unflagged version as well

- `views/tables/PositionsTable/PositionsTriggersCell` -> renamed to `PositionsTriggersCellDeprecated`

- New: `PositionsTriggersCell`
  - Only rendered in ui refresh
  - Fixed width 70px for SL and TP boxes, can adjust
  - Verified that "View more" and prices truncate when necessary

- `tables/PositionsTable/PositionsActionsCell`
  - Pass through optional `--toolbar-margin` adjustment
  - Update button styling when uiRefresh is on

- `tables/PositionsTable/PositionsMarginCell`
  - Custom styling on margin cell when ui refresh is enabled

- `tables/PositionsTable`
  - Update some existing keys + renderings; 

## Components

- `<Output>`
  - replace en dash with em dash

- `<Table>`
  - Add optional `align` override for columns (I couldn't figure out the typing here tho rip my `.value as any`)

- `Table/ActionsTableCell`
  - Update to not render trigger edit button when `uiRefresh` enabled
  - Updated icon sizing for when `uiRefresh` enabled + remove button backgrounds + borders
  
---

### CODE REVIEW

File: src/components/Output.tsx [hunk 1]
```diff
@@ -469,7 +469,7 @@ const $Text = styled.output<{ withParentheses?: boolean }>`
     opacity: 0.5;
 
     &:after {
-      content: '-' var(--output-afterString);
+      content: '—' var(--output-afterString);
```
Comment (aforaleka) [inline src/components/Output.tsx:472]: .___, hahahha what changed lmao

File: src/views/tables/PositionsTable/PositionsTriggersCell.tsx [hunk 2]
```diff
@@ -91,84 +93,44 @@ export const PositionsTriggersCell = ({
...
-              {stringGetter({ key: STRING_KEYS.EDIT_STOP_LOSS })}
-            </Button>
-          }
-          slotTrigger={triggerButton}
-        />
-      ) : (
-        triggerButton
-      );
-    };
+  const editButton = (
```
Comment (aforaleka) [inline src/views/tables/PositionsTable/PositionsTriggersCell.tsx:109]: suggestion -- can we actually make the whole cell clickable to open the triggers dialog? (esp. when empty / no view-orders button); currently the target of the edit button seems kinda small and i didn't click it on my first try

**Review by aforaleka — APPROVED** [review]
just nits / followup improvements!

File: src/views/tables/PositionsTable/PositionsMarginCell.tsx [hunk 3]
```diff
@@ -65,7 +88,16 @@ export const PositionsMarginCell = ({ position }: PositionsMarginCellProps) => {
     </TableCell>
   );
 };
+
 const $EditButton = styled(IconButton)`
+  --button-textColor: var(--color-text-0);
+  --button-hover-textColor: var(--color-text-1);
+  --button-backgroundColor: transparent;
+  --button-border: none;
+  --button-width: min-content;
```
[the remark this answers is not in the document]
Comment (aforaleka) [reply src/views/tables/PositionsTable/PositionsMarginCell.tsx:97]: actually i have pr that does this!

File: src/components/Output.tsx (hunk 1 above)
Comment (moo-onthelawn) [reply src/components/Output.tsx:472]: loool I think it renders as slightly larger on the web 😭 my old designer used to be super opinionated about this

File: src/views/tables/PositionsTable/PositionsMarginCell.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (moo-onthelawn) [reply src/views/tables/PositionsTable/PositionsMarginCell.tsx:97]: omg mvp ty

File: src/views/tables/PositionsTable/PositionsTriggersCell.tsx (hunk 2 above)
Comment (moo-onthelawn) [reply src/views/tables/PositionsTable/PositionsTriggersCell.tsx:109]: ah ok will follow up to add in a seperate PR!! gonna merge ahead into parent branch tho for sake of not bothering for re-review

**Pull request merged**